src/VisualStudio/Core/Def/Telemetry/VisualStudioWorkspaceTelemetryService.ProjectTelemetry.cs

```diff
@@ -41,71 +33,19 @@ internal class VisualStudioProjectTelemetryService
         private const string TelemetryDocumentsCountName = PropertyPrefix + "Documents.Count";
         private const string TelemetryAdditionalDocumentsCountName = PropertyPrefix + "AdditionalDocuments.Count";
 
-        private readonly VisualStudioWorkspaceImpl _workspace;
-
         /// <summary>
         /// Our connection to the remote OOP server. Created on demand when we startup and then
         /// kept around for the lifetime of this service.
         /// </summary>
         private RemoteServiceConnection<IRemoteProjectTelemetryService>? _lazyConnection;
 
-        /// <summary>
-        /// Queue where we enqueue the information we get from OOP to process in batch in the future.
-        /// </summary>
-        private readonly AsyncBatchingWorkQueue<ProjectTelemetryData>? _workQueue;
-
-        [ImportingConstructor]
-        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-        public VisualStudioProjectTelemetryService(
-            VisualStudioWorkspaceImpl workspace,
-            IThreadingContext threadingContext) : base(threadingContext)
-        {
-            _workspace = workspace;
-
-            _workQueue = new AsyncBatchingWorkQueue<ProjectTelemetryData>(
-                TimeSpan.FromSeconds(1),
-                NotifyTelemetryServiceAsync,
-                threadingContext.DisposalToken);
-        }
-
         public void Dispose()
         {
             _lazyConnection?.Dispose();
         }
 
-        void IEventListener<object>.StartListening(Workspace workspace, object _)
-        {
-            if (workspace is VisualStudioWorkspace)
-                _ = StartAsync();
-        }
-
-        private async Task StartAsync()
+        private async Task StartProjectTelemetryWorkerAsync(RemoteHostClient client, CancellationToken cancellationToken)
```

Comment:
❗ This is an O(solution size) telemetry operation, and therefore is not zero-overhead. It cannot be enabled in its current form.